streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -557,7 +577,9 @@ public Set<Task> getRemovedTasks() {
     }
 
     public Set<Task> getPausedTasks() {
-        return Collections.unmodifiableSet(new HashSet<>(pausedTasks.values()));
+        return stateUpdaterThread != null
```

Comment:
We can only pause tasks with the state updated thread? What if they are using the old/different implementation? I think that can get confusing